test/Table_Tests/src/IO/Fetch_Spec.enso

```diff
@@ -94,3 +103,330 @@ add_specs suite_builder =
+                Data.fetch url1
+
+        group_builder.specify "Should not cache Data.download" pending=pending_has_url <| Test.with_retries <|
+            target_file = enso_project.data / "transient" / "my_download0.txt"
+
+            HTTP.clear_response_cache
+            target_file.delete_if_exists
+
+            Data.download url0 target_file
+            get_num_response_cache_entries . should_equal 0
+
+            target_file.delete_if_exists
+
+        group_builder.specify "Data.download is not affected by caching limits" pending=pending_has_url <| Test.with_retries <|
+            target_file = enso_project.data / "transient" / "my_download0.txt"
+            Panic.with_finalizer reset_size_limits <|
+                EnsoHTTPResponseCache.setMaxTotalCacheSizeOverrideTestOnly 120
+                EnsoHTTPResponseCache.setMaxFileSizeOverrideTestOnly 100
+                Data.download base_url_with_slash+"test_download?length=200" target_file
+                target_file.read.length . should_equal 200
+                target_file.delete_if_exists
+
+        group_builder.specify "Should not cache for methods other than GET" pending=pending_has_url <| Test.with_retries <|
+            HTTP.clear_response_cache
+
+            expect_counts [0, 0] <|
+                Data.post url_post (Request_Body.Text "hello world")
+
+        group_builder.specify "HTTP request with a non-GET method should reject a Cache_Policy argument" pending=pending_has_url <| Test.with_retries <|
+            request = Request.new HTTP_Method.Post url_post [] Request_Body.Empty
+            HTTP.new.request request cache_policy=Cache_Policy.Use_Cache . should_fail_with Illegal_Argument
+
+        group_builder.specify "Should be able to clear caches" pending=pending_has_url <| Test.with_retries <|
+            HTTP.clear_response_cache
+            Data.fetch url0
+            get_num_response_cache_entries . should_equal 1
+            HTTP.clear_response_cache
+            get_num_response_cache_entries . should_equal 0
+
+        group_builder.specify "Cache key should only dependon the URI and headers" pending=pending_has_url <| Test.with_retries <|
+            HTTP.clear_response_cache
+            expect_counts [0, 1] <|
+                Data.fetch url0
+                Data.fetch url0
+            expect_counts [1, 2] <|
+                Data.fetch url1
+                Data.fetch url1
+
+        group_builder.specify "Cache key should depend on the URI" pending=pending_has_url <| Test.with_retries <|
+            HTTP.clear_response_cache
+            expect_counts [0, 2] <|
+                Data.fetch url0
+                Data.fetch url1
+
+        group_builder.specify "Cache key should depend on the headers" pending=pending_has_url <| Test.with_retries <|
+            HTTP.clear_response_cache
+            expect_counts [0, 2] <|
+                Data.fetch url0 headers=headers0
+                Data.fetch url0 headers=headers1
```

Comment:
Also here, what do these 2 tests add in comparison to the test above?

They could be passed if the cache saved _every_ response as separate cache key.

IMO these redundant test cases do not add much value.

Reply:
I'll keep the headers one, but added extra fetches to the same url, to avoid the false negative you mentioned.